app/ide-desktop/lib/client/src/preload.ts

```diff
@@ -50,3 +62,28 @@ contextBridge.exposeInMainWorld('enso_console', {
     // Print an error message with `console.error`.
     error: (data: any) => ipcRenderer.send('error', data),
 })
+
+
+
+// ==========================
+// === Authentication API ===
+// ==========================
+
+/** Exposes an object on the main window; object is used by the dashboard to:
+ * - open OAuth flows in the system browser, and
+ * - handle deep links from the system browser or email client to the dashboard. */
+contextBridge.exposeInMainWorld(AUTHENTICATION_API_KEY, {
+    /** Opens a URL in the system browser (rather than in the app).
+     *
+     * OAuth URLs must be opened this way because the dashboard application is sandboxed and thus
+     * not privileged to do so unless we explicitly expose this functionality. */
+    openUrlInSystemBrowser: (url: string) => ipcRenderer.send(ipc.channel.openUrlInSystemBrowser, url),
+    /** Set the callback that will be called when a deep link to the application is opened.
+     *
+     * The callback is intended to handle links like
+     * `enso://authentication/register?code=...&state=...` from external sources like the user's
+     * system browser or email client. Handling the links involves resuming whatever flow was in
+     * progress when the link was opened (e.g., an OAuth registration flow). */
+    setDeepLinkHandler: (callback: (url: string) => void) =>
```

Comment:
this is not used anyhweher in the code. I understand that it will be used by the subsequent PR, right?

Reply:
This will be used by the subsequent PR. The dashboard is responsible for setting a handler, since it's the module that knows where/how to route the authentication pages.